### PR TITLE
add oauth token introspection

### PR DESCRIPTION
**Pull request by calellowitz**
This sets commcare-connect to use connect id as a fallback server for checking OAuth tokens, and add oauth to the allowed authentication methods.

### CODE REVIEW

**Comment by calellowitz** [comment]
Sorry, thought I had requested review on this a while ago. Not sure how I missed that

**Review by snopoke — APPROVED** [review]

File: commcare_connect/users/models.py [hunk 1]
```diff
@@ -13,12 +14,24 @@ class User(AbstractUser):
...
+        _("username"),
+        max_length=150,
+        unique=True,
+        help_text=_("Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only."),
+        validators=[username_validator],
+        error_messages={
+            "unique": _("A user with that username already exists."),
+        },
+        null=True,
+    )
```
Comment (snopoke) [inline commcare_connect/users/models.py:34]: Is this just copied from `AbstractUser`? Does removing `username = None` not have the same effect?

**Review by snopoke — APPROVED** [review]

File: commcare_connect/users/models.py (hunk 1 above)
Comment (calellowitz) [reply commcare_connect/users/models.py:34]: It is copied but it adds `null=True` since email enrolled users don't have a username.

**Comment by calellowitz** [comment]
@snopoke I think I got to all your comments but let me know if that is not true

File: commcare_connect/users/tests/test_views.py [hunk 2]
```diff
@@ -106,3 +108,19 @@ def test_not_authenticated(self, user: User, rf: RequestFactory):
...
+
+    def test_view(self, mobile_user: User, rf: RequestFactory):
+        request = rf.post("/fake-url/", data={"commcare_username": "abc", "connect_username": mobile_user.username})
+        request.user = mobile_user
+        with mock.patch(
+            "oauth2_provider.views.mixins.ClientProtectedResourceMixin.authenticate_client"
+        ) as authenticate_client:
+            authenticate_client.return_value = True
+            response = create_user_link_view(request)
+            print(response)
```
Comment (snopoke) [inline commcare_connect/users/tests/test_views.py:123]: nit: print

**Review by snopoke — APPROVED** [review]

**Pull request merged**